src/main/java/com/adyen/httpclient/HttpURLConnectionClient.java

```diff
@@ -232,41 +242,47 @@ private String doPostRequest(HttpURLConnection httpConnection, String requestBod
         if (connection instanceof HttpsURLConnection) {
             HttpsURLConnection httpsConnection = (HttpsURLConnection) connection;
 
-//            try {
-//                // Create a trust manager that does not validate certificate chains
-//                TrustManager[] trustAllCerts = new TrustManager[] {new X509TrustManager() {
-//                    public X509Certificate[] getAcceptedIssuers() {
-//                        return null;
-//                    }
-//
-//                    public void checkClientTrusted(X509Certificate[] certs, String authType) {
-//                    }
-//
-//                    public void checkServerTrusted(X509Certificate[] certs, String authType) {
-//                    }
-//                }
-//                };
-//
-//                // Install the all-trusting trust manager
-//                SSLContext sc = SSLContext.getInstance("SSL");
-//                sc.init(null, trustAllCerts, new java.security.SecureRandom());
-//                httpsConnection.setSSLSocketFactory(sc.getSocketFactory());
-//            } catch (NoSuchAlgorithmException | KeyManagementException e) {
-//                e.printStackTrace();
-//            }
+            try {
+                // Create new KeyStore for the terminal certificate
+                InputStream certificateInput = new FileInputStream(terminalCertificatePath);
+                CertificateFactory certificateFactory = CertificateFactory.getInstance("X.509");
+                X509Certificate cert = (X509Certificate) certificateFactory.generateCertificate(certificateInput);
+
+                KeyStore keyStore = KeyStore.getInstance("JKS");
+                keyStore.load(null, null);
+                keyStore.setCertificateEntry("TerminalCertificate", cert);
+
+                TrustManagerFactory trustFactory =
+                        TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+                trustFactory.init(keyStore);
+                TrustManager[] trustManagers = trustFactory.getTrustManagers();
+
+                // Install the terminal certificate trust manager
+                SSLContext sc = SSLContext.getInstance("SSL");
+
+                sc.init(null, trustManagers, new java.security.SecureRandom());
+                httpsConnection.setSSLSocketFactory(sc.getSocketFactory());
+            } catch (GeneralSecurityException | IOException e) {
+                throw new HTTPClientException("Error loading certificate from path", e);
+            }
 
             // Create terminal-trusting host name verifier
             HostnameVerifier terminalHostsValid = new HostnameVerifier() {
                 public boolean verify(String host, SSLSession session) {
-                    if (host.matches(".+\\..+\\.terminal\\.adyen\\.com")) {
+                    String url = host + ":" + session.getPeerPort();
+                    if (url.matches(TERMINAL_API_HOST_REGEX) || url.matches(TERMINAL_API_IP_REGEX)) {
```

Comment:
let's validate this as well with @AlexandrosMor I am not sure this is right.

Reply:
Made the change to skip this validation entirely, like we talked earlier @rikterbeek @AlexandrosMor